### PR TITLE
Implement Schedule Hierarchy

### LINKED ISSUE #190: Implement Schedule Hierarchy
Problem statement:
There is no schedule in the project so far, so there is a necessity to be able to schedule things properly. There will be a need for various schedules.

Proposed solution:
Create a Schedule base class. We can implement DriverSchedule, RestaurantSchedule, DeliverySlot subclasses. 

Affected models:
The Schedule base class is added, with DriverSchedule, RestaurantSchedule, and DeliverySlot subclasses to extend it.

Acceptance criteria:
Testing (time validation & availability) done with Mocha/Chai.
Validation is in scheduling code.
Subclasses extend schedule.

### PR DESCRIPTION
**Pull request by KaiFu7k**
## Overview
This PR implements a polymorphic `scheduleModel` hierarchy to handle different timing and availability requirements across the application. It includes a base class, three distinct subclasses, a suite of automated unit tests, and a responsive frontend demonstration.

- **Linked Issue:** Fixes #190 
- **Feature Branch:** `feature/schedule-hierarchy`

## Polymorphic Implementation
I have implemented the following class structure in `core/models/users/scheduleModel.js`:
- **Parent Class:** `ScheduleModel` - Defines the base properties and the polymorphic `isAvailable()` method.
- **Subclass 1: `RestaurantSchedule`** - Overrides `isAvailable()` to check if a specific time falls within opening and closing hours.
- **Subclass 2: `DriverSchedule`** - Overrides `isAvailable()` to account for shift ranges and the driver's "break" status.
- **Subclass 3: `DeliverySlot`** - Overrides `isAvailable()` using a boolean state to track whether a specific slot has been booked.

## Automated Testing (Mocha/Chai)
I have implemented 4 meaningful test cases in `test/schedule.test.js` to validate the full class hierarchy and ensure polymorphic behavior:

1. **Base Class Behavior (Safety Net):** Validates that the parent `ScheduleModel` returns a default value of `false`. This ensures the system fails safely and satisfies the rubric requirement for validating base class logic.
2. **Subclass Override (Restaurant):** A "Happy Path" test that validates the time-range logic implemented in `RestaurantSchedule`.
3. **Subclass Override (Driver):** A behavioral validation test for `DriverSchedule` ensuring that the `isAvailable()` method correctly prioritizes the "onBreak" status over shift hours.
4. **State-Based Edge Case (Delivery Slot):** Verifies the state-transition logic for `DeliverySlot`, confirming that the same polymorphic method correctly handles boolean booking states instead of just time integers.
5. **Constructor Validation (Failure Case):** Confirms that subclasses throw meaningful errors when initialized with invalid logical data (e.g., opening time set later than closing time). This ensures data integrity and addresses feedback regarding missing input validation.
**CI Status:** GitHub Actions is configured and all tests are passing (see checkmark below).

## Responsive Webpage
A minimal, responsive webpage has been created at `pages/schedules/schedule.html`:
- **Framework:** Bootstrap 5 (Responsive for Mobile/Desktop).
- **Theme:** Official CSW color scheme (in the styles.css).
- **Functionality:** Demonstrates the unique logic of all three subclasses through an interactive UI.

## How to Test
1. **Backend:** Run `npm install` and then `npm test` in the terminal to verify the 4 Mocha tests pass.
2. **Frontend:** Open `pages/schedules/schedule.html` in a browser. Use the interactive cards to test store hours, toggle driver breaks, and book delivery slots.

## Screenshots
On computer:
homepage forcefoodgood desktop

On mobile:
homepage forcefoodgood mobile

### CODE REVIEW

File: core/models/scheduleModel.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    constructor(id) {
+        this.id = id;
+    }
+    isAvailable(param) {
+        console.log("Checking for the general availability...");
+        return false;
+    }
+}
+
+//first subclass with the restaurant
```
Comment (david-safro) [inline core/models/scheduleModel.js]: RestaurantSchedule and DriverSchedule accept openTime/closeTime and shiftStart/shiftEnd but don't validate them. If someone passes openTime > closeTime, isAvailable() will silently return false for all times with no error.
